### ISSUE TITLE
Nightly deployment

### ISSUE THREAD

**Issue by LorenzoBettini** [xtext/xtext-monorepo#11]

**Comment by LorenzoBettini** [comment]
@cdietrich I'd like to propose the following next steps now that we have a JIRO job deploying snapshots on Sonatype and a nightly p2 repo:

- disable the old job for nightly deployment, and make the new job the nightly one; the nightly p2 monorepo can be used now for integration tests
- try to make/simulate a milestone (Beta1? or M1): on sonatype, the artifacts will be staged, so unless manually released, they will not be official; the milestone update site would be in the monorepo subdirectory (so not in the official Xtext milestone composite repository).

**Comment by szarnekow** [comment]
@LorenzoBettini would it interfere with your plans if you or  I push the monorepo/main branch to eclipse/xtext/main already?

**Comment by LorenzoBettini** [comment]
@szarnekow Would that make the monorepo sources "official"? I mean: I should then go on working on that repository, shouldn't I?
First, I'd like to merge https://github.com/xtext/xtext-monorepo/tree/lb_issue_11_nightly_deployment, which I've just done.
Would the `main` branch be disconnected from the `master` branch? If they're disconnected and the `main` branch is set as the real base branch (see the setting below), I guess cloning will only clone the `main` branch by default and the separate old `master` branch will be ignored by default; but I'm not sure about that.

In any case, you can go right ahead: I'll stop working on that until you finished. I've pushed to `main` all my work. 

![image](https://user-images.githubusercontent.com/1202254/227730322-091feff5-66b0-4d3b-b04f-137d68953dd7.png)

**Comment by cdietrich** [comment]
we need to open a ticket with it to change the default branch.
but i assume we can "treat" it as such anyway

**Comment by LorenzoBettini** [comment]
> we need to open a ticket with it to change the default branch. but I assume we can "treat" it as such anyway

Yes, but if they are completely separate, the default branch will be cloned by default.
I don't remember whether you planned to separate the two branches.

**Comment by szarnekow** [comment]
The main branch is setup such that is completely disconnected from eclipse/xtext/master such that we can delete all the old commits from eclipse/xtext. That content was previously archived to eclipse/xtext-archive.

**Comment by cdietrich** [comment]
sebastian has opened https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/2864

**Comment by szarnekow** [comment]
> I mean: I should then go on working on that repository, shouldn't I?

Yes, from this moment on this will avoid an indirection. If feasible please perform the next steps directly in eclipse/xtext

**Comment by cdietrich** [comment]
@LorenzoBettini will you update the jobs to consume from there?

**Comment by LorenzoBettini** [comment]
> @LorenzoBettini will you update the jobs to consume from there?

I've already updated this one https://ci.eclipse.org/xtext/view/Xtext-release/job/xtext-monorepo-full-deploy-nightly

Now we should create another multibranch pipeline job I guess?
Should I do that right now?
Since the `Jenkinsfile` is not present in the current `master` it should be safe to do so.

I would have expected that GitHub Actions workflow would have started on https://github.com/eclipse/xtext even if not on the default branch... let's see... maybe it takes some time or simply the default branch must be first updated.

**Comment by cdietrich** [comment]
have no idea. i assume it will work with any branch

**Comment by LorenzoBettini** [comment]
> have no idea. i assume it will work with any branch

I meant: it should not try to build all the current branches. It should build only the ones with a Jenkinsfile. Let's try.

**Comment by cdietrich** [comment]
yes so we might delay until we have gotten rid of the other branches.
side question: you did not setup/cleanup the other target platform yet correct?

**Comment by LorenzoBettini** [comment]
@cdietrich I tried to create the multibranch job:

![image](https://user-images.githubusercontent.com/1202254/227731828-f69040c2-5ccc-4d69-bbf3-da2be7a66cf8.png)

It had tried to build other branches because I guess there was a Jenkinsfile there.
I stopped all those builds but the build for `main`.

Should I disable the multiline job for the moment?

**Comment by cdietrich** [comment]
yes

**Comment by LorenzoBettini** [comment]
OK, I disabled that for the moment.
The job for `main` can go on for the moment.
I checked the other jobs that started and I confirm that I aborted them before they did any damage ;)

**Comment by LorenzoBettini** [comment]
This one https://ci.eclipse.org/xtext/view/Xtext-release/job/xtext-monorepo-full-deploy-nightly which is currently running is still publishing the nightly update site on https://download.eclipse.org/modeling/tmf/xtext/monorepo/updates/nightly/

**Comment by LorenzoBettini** [comment]
> side question: you did not setup/cleanup the other target platform yet correct?

Sorry, I forgot to answer this one: no, I haven't.

**Comment by LorenzoBettini** [comment]
> I guess cloning will only clone the `main` branch by default and the separate old `master` branch will be ignored by default; but I'm not sure about that.

I see that now `main` is the default branch. Unfortunately, cloning still clones everything even the old `master` and its huge history.
